x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycle.java

```diff
@@ -179,6 +179,7 @@ public Collection<Object> createComponents(
             xContentRegistry
         );
         ilmTemplateRegistry.initialize();
+        components.add(ilmTemplateRegistry);
```

Comment:
Bookkeeping note: As mentioned in my comment in the APMPlugin class, if we move the APMRolloverIT to core, this doesn't need to be exposed.